src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelBindingContext.cs

```diff
@@ -174,5 +178,11 @@ public ModelBindingContext()
         /// is eligible for model binding.
         /// </summary>
         public Func<ModelBindingContext, string, bool> PropertyFilter { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="ValidationStateDictionary"/>. Used for tracking validation state to
+        /// customize validation behavior for a model object.
+        /// </summary>
+        public ValidationStateDictionary ValidationState { get; set; }
```

Comment:
Why does this property have a setter? Is this just for the odd special-case almost-a-copy-constructor code in `CompositeModelBinder`?